libdnf/comps/group/group.cpp

```diff
@@ -88,75 +88,80 @@ std::string Group::get_description() const {
 std::string Group::get_translated_name(const char * lang) const {
     libdnf::solv::Pool & pool = get_pool(base);
 
-    std::string translation;
+    // Go through all group solvables and return first translation found.
     for (GroupId group_id : group_ids) {
-        Solvable * solvable = pool->solvables + group_id.id;
-        if (solvable_lookup_str_lang(solvable, SOLVABLE_SUMMARY, lang, 1)) {
-            translation = solvable_lookup_str_lang(solvable, SOLVABLE_SUMMARY, lang, 1);
-            // Return translation only if it's different from the untranslated string.
-            if (translation != solvable_lookup_str(solvable, SOLVABLE_SUMMARY)) {
-                return translation;
+        Solvable * solvable = pool.id2solvable(group_id.id);
+        if (const char * translation = solvable_lookup_str_lang(solvable, SOLVABLE_SUMMARY, lang, 1)) {
+            // Return translation only if it's different from the untranslated string
+            // (solvable_lookup_str_lang returns the untranslated string if there is no translation).
+            const char * untranslated = solvable_lookup_str(solvable, SOLVABLE_SUMMARY);
+            if (translation != untranslated and strcmp(translation, untranslated) != 0) {
+                return std::string(translation);
             }
         }
     }
+    // If no translation was found, return the untranslated string.
     return this->get_name();
 }
 
 
 // TODO(pkratoch): Test this
 std::string Group::get_translated_name() const {
-    libdnf::solv::Pool & spool = get_pool(base);
-    Pool * pool = *spool;
+    libdnf::solv::Pool & pool = get_pool(base);
 
-    std::string translation;
+    // Go through all group solvables and return first translation found.
     for (GroupId group_id : group_ids) {
-        Solvable * solvable = pool->solvables + group_id.id;
-        if (solvable_lookup_str_poollang(solvable, SOLVABLE_SUMMARY)) {
-            translation = solvable_lookup_str_poollang(solvable, SOLVABLE_SUMMARY);
-            // Return translation only if it's different from the untranslated string.
-            if (translation != solvable_lookup_str(solvable, SOLVABLE_SUMMARY)) {
-                return translation;
+        Solvable * solvable = pool.id2solvable(group_id.id);
+        if (const char * translation = solvable_lookup_str_poollang(solvable, SOLVABLE_SUMMARY)) {
+            // Return translation only if it's different from the untranslated string
+            // (solvable_lookup_str_lang returns the untranslated string if there is no translation).
+            const char * untranslated = solvable_lookup_str(solvable, SOLVABLE_SUMMARY);
+            if (translation != untranslated and strcmp(translation, untranslated) != 0) {
```

Comment:
Please use `&&` instead of `and`. Used multiple times in this file.
C++ supports alternative operators https://en.cppreference.com/w/cpp/language/operator_alternative , but we don't use them in the libdnf project.
Especially `and` and `or` are used to adapt to historical restrictive character sets.

Reply:
Thanks, fixed. And I found one more instance in `include/libdnf/comps/group/group.hpp`.